### PR TITLE
Cleanup code of past onboarding experiment and prototype mentions

### LINKED ISSUE #6333: Cleanup codebase from expired onboarding experiment and prototype mentions
### Description
This is a follow-up issue from #6298. In there, we focused on removing the mentions of 'prototype' that were user-facing. That included modifying the URL and title of the page and making sure that a user won't find anything inherited from the prototype version.

However, the codebase has still multiple references to "prototype", like `client/onboarding-prototype`. If there is any conflict with the current onboarding area and we can't use "Onboarding" to mention it, we should rename it using the nomenclature of #6298, so instead of "Onboarding Prototype", it should be "Onboarding Flow". Eventually, if the new flow is successful, will replace the whole old onboarding and an additional clean-up will occur there.

This issue will also cover the clean-up of the already expired onboarding experiment introduced in #4091.

We need to be extra careful with `includes/admin/class-wc-rest-payments-onboarding-controller.php`, as that defined controller for the REST route `payments/onboarding` has a mix of areas of the new `onboarding-flow` mixed within it.

### PR DESCRIPTION
**Pull request by daquinons**
Fixes #6333 

#### Changes proposed in this Pull Request

Being a followup of https://github.com/Automattic/woocommerce-payments/issues/6298, this issue continues some of what was started there:

- The onboarding experiment introduced in #4091 is removed. Prior to this PR, it was living in `/onboarding` path.
- From that past experiment, there are still two methods that I've chosen to not touch. Those are `get_business_types` and `get_required_verification_information` in [here](https://github.com/Automattic/woocommerce-payments/blob/fix/6333-cleapup-onboarding-experiment/includes/admin/class-wc-rest-payments-onboarding-controller.php#L67). Even though those features are not used in the newest onboarding I think they will potentially be in the future in the fully owned one.
- The new onboarding flow is moved to `/onboarding` instead of `/onboarding-flow` that was set in #6298 
- Code references to `Prototype` in CSS classes and React components are removed

#### Testing instructions
The main goal of the testing instructions is to make sure that nothing broke after these changes :D

**Creation of PO account**

* Check out this branch, and use the latest `trunk` on the server.
* Enable Progressive Onboarding in the dev tools settings, and ensure you don't have an active connected account on your client.
* Go through the onboarding using the values necessary to onboard a PO account (US, Individual, Less than 250k, Ready to start selling).
* Check the onboarding flow path parameter is `payments/onboarding` when clicking on "Finish setup", such as http://localhost:8082/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Fonboarding
* Complete the onboarding, it should complete successfully and behave as normal afterwards. You should be taken to the WCPay dashboard and prompted to either enable deposits or proceed with PO.

**Creation of non-PO account with the new flow**

* Repeat the steps above, but this time onboard an account which is not PO (for example, select a different country in the onboarding).
* The path should still be `payments/onboarding` when clicking on "Finish setup", such as http://localhost:8082/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Fonboarding
* The account will onboard successfully.

**Testing the old flow**

* Disable `Enable Progressive Onboarding` in the dev tools.
* Perform an onboarding - it should work exactly how it did previously, user will be taken to the Stripe KYC and will be prompted to enter their country information and all other details necessary to onboard.
* Onboarding will complete successfully.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6564` or branch name `fix/6333-cleapup-onboarding-experiment` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=fix/6333-cleapup-onboarding-experiment) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 157706316bfc4fced98fec0d2c9ea97849b76224
- Build time: 2023-06-28 09:08:37 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by daquinons** [comment]
After writing the explanation of why I was not touching the methods in [here](https://github.com/Automattic/woocommerce-payments/blob/fix/6333-cleapup-onboarding-experiment/includes/admin/class-wc-rest-payments-onboarding-controller.php#L67) I started to think if I should have kept around also the files from `onboarding-experiment`... 🤔 Being all well isolated in the same folder, we can always recover them in the future from the history, I guess.

File: client/index.js [hunk 1]
```diff
@@ -58,7 +57,7 @@ addFilter(
 		} );
 
 		pages.push( {
-			container: OnboardingExperimentPage,
+			container: Onboarding,
```
Comment (ismaeldcom) [inline client/index.js]: _Nitpick_: To keep it consistent we can name it `OnboardingPage` rather than `Onboarding`

**Review by ismaeldcom — APPROVED** [review]
Thanks for taking care of this clean up! Testing instructions work as expected, also tested the experiment case.

I just left a tiny nitpick, but it's ready to go 🚢

File: client/index.js (hunk 1 above)
Comment (daquinons) [reply client/index.js]: That is actually right! https://github.com/Automattic/woocommerce-payments/pull/6564/commits/8609391e18129f62d18ef4b15add5ee72520e2c6

**Pull request merged**